sympy/external/ntheory.py

```diff
@@ -50,6 +50,34 @@ def bit_scan0(x, n=0):
     return bit_scan1(x + (1 << n), n)
 
 
+def remove(x, f):
+    if f < 2:
+        raise ValueError("factor must be > 1")
+    if x == 0:
+        return 0, 0
+    if f == 2:
+        b = bit_scan1(x)
+        return x >> b, b
+    m = 0
+    y, rem = divmod(x, f)
+    while not rem:
+        x = y
+        m += 1
+        if m > 5:
+            pow_list = [f**2]
```

Comment:
This is an interesting variation of what we had before but it now involves a list. Is there a reason to prefer this over to keeping track of the scalar `e`? One thing I see is that you only work with squares while the other code worked with even exponents.

Reply:
The current version calls `multiplicity` recursively. To stop the recursion, I use the list.

Also, the current version uses only powers of 2. The `e` will never be 6, 10, etc.

Reply:
Thanks.